src/metamodels/loom_metamodel.py

```diff
@@ -0,0 +1,905 @@
+            (generator_id, num_models)
+            VALUES (?, ?)
+        ''', (generator_id, len(modelnos) + num_existing))
+
+    def _get_num_models(self, bdb, generator_id):
+        return util.cursor_value(bdb.sql_execute('''
+            SELECT num_models FROM bayesdb_loom_generator_model_info
+            WHERE generator_id = ?;
+        ''', (generator_id,)))
+
+    def drop_generator(self, bdb, generator_id):
+        self._del_cache_entry(bdb, generator_id, None)
+
+        with bdb.savepoint():
+            self.drop_models(bdb, generator_id)
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_generator
+                WHERE generator_id = ?
+            ''', (generator_id,))
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_generator_model_info
+                WHERE generator_id = ?
+            ''', (generator_id,))
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_string_encoding
+                WHERE generator_id = ?
+            ''', (generator_id,))
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_column_ordering
+                WHERE generator_id = ?
+            ''', (generator_id,))
+
+    def drop_models(self, bdb, generator_id, modelnos=None):
+        with bdb.savepoint():
+            name = self._get_name(bdb, generator_id)
+            if modelnos is not None:
+                raise BQLError(bdb, 'Loom cannot drop specific models numbers.')
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_column_kind_partition
+                WHERE generator_id = ?;
+            ''', (generator_id,))
+            bdb.sql_execute('''
+                DELETE FROM bayesdb_loom_row_kind_partition
+                WHERE generator_id = ?;
+            ''', (generator_id,))
+            q_server = self._get_cache_entry(bdb, generator_id, 'q_server')
+            if q_server is not None:
+                q_server.close()
+            preql_server = self._get_cache_entry(bdb, generator_id, 'preql_server')
+            if preql_server is not None:
+                preql_server.close()
+            self._del_cache_entry(bdb, generator_id, 'q_server')
+            self._del_cache_entry(bdb, generator_id, 'preql_server')
+
+            bdb.sql_execute('''
+                UPDATE bayesdb_loom_generator_model_info
+                SET num_models = 0
+                WHERE generator_id = ?
+            ''', (generator_id,))
+            name = self._get_name(bdb, generator_id)
+            paths = loom.store.get_paths(name)
+            if 'root' in paths:
+                folder_with_models = os.path.join(paths['root'], 'samples')
+                os.system("rm -rf {}".format(folder_with_models))
+
+    def analyze_models(self, bdb, generator_id, modelnos=None, iterations=1,
+            max_seconds=None, ckpt_iterations=None, ckpt_seconds=None,
+            program=None):
```

Comment:
Resolved. It turns out that the loom variables such as "VERBOSITY" are set at the time loom is imported. Therefore, changing the environment variables from within loom_metamodel (after loom has already been imported) doesn't do anything to loom itself because it has already read the environment variable. In order for the analyze_models() function in loom_metamodel to change the verbosity level, it would have to set the environment variable 'LOOM_VERBOSITY' to 0, reimport loom, and then use the newly imported loom when calling loom.tasks.infer. This seems like too much of a hack so I just made analyze_models() raise a BQLError if a program is provided.